Updated the intro to the session pool limit article (#11325)
ydb-platform/ydb: ydb/docs/en/core/recipes/ydb-sdk/session-pool-limit.md

{% include [work in progress message](_includes/addition.md) %}

> {{ ydb-short-name }} creates an [actor](../../concepts/glossary.md#actor) for each session. Consequently, session pool size of a client affects resource consumption (RAM, CPU) on the server side of {{ ydb-short-name }}.

Reviewer: ```suggestion
{{ ydb-short-name }} creates an [actor](../../concepts/glossary.md#actor) for each session. Consequently, the session pool size of a client affects resource consumption (RAM, CPU) on the server side of {{ ydb-short-name }}.
```

> For example, if 1000 clients of the same database have 1000 sessions each, 1000000 [actors](../../concepts/glossary.md#actor) are created on the server side, and this number of actors consumes a significant amount of memory and CPU. If you do not limit the number of sessions on the client, this may result in a slow cluster that is close to a failure.

Reviewer: ```suggestion
For example, if 1000 clients of the same database have 1000 sessions each, then 1000000 [actors](../../concepts/glossary.md#actor) are created on the server side, consuming a significant amount of memory and CPU. If you do not limit the number of sessions on the client side, this may result in a slow cluster that is at risk of failure.
```

> By default, the {{ ydb-short-name }} SDK has a limit of 50 sessions when using native drivers. There is no limit for third-party libraries, such as Go database/sql.

Reviewer: ```suggestion
By default, the {{ ydb-short-name }} SDK sets a limit of 50 sessions when using native drivers. However, no session limit is enforced for third-party libraries, such as Go `database/sql`.
```

> A good recommendation is to set the limit on the number of client sessions to the minimum required for the normal operation of the client app. Keep in mind that sessions are single-threaded both on the server and client side. So if the application needs to make 1000 simultaneous (inflight) requests to {{ ydb-short-name }} for its estimated load, the limit should be set to 1000 sessions.
> 
> Here it's necessary to distinguish between the estimated RPS (requests per second) and inflight. In the first case, this is the total number of requests to {{ ydb-short-name }} completed within 1 second. For example, if RPS=10000 and the average latency is 100&nbsp;ms, it's sufficient to set the session limit to 1000. This means that each session will perform an average of 10 consecutive requests for the estimated second.

Reviewer: ```suggestion
It’s recommended to set the client session limit to the minimum required for the normal operation of the client application. Keep in mind that sessions are single-threaded on both the server and client sides. For instance, if the application needs to handle 1000 simultaneous (in-flight) requests to {{ ydb-short-name }} based on its estimated load, the session limit should be set to 1000.

It is important to distinguish between estimated RPS (requests per second) and in-flight requests. RPS refers to the total number of requests completed by {{ ydb-short-name }} within one second. For example, if RPS = 10000 and the average latency is 100 ms, a session limit of 1000 is sufficient. This configuration allows each session to perform an average of 10 consecutive requests within the estimated second.
```
